src/features/metadata-based-view/MetadataQueryAPIHelper.js

```diff
@@ -54,18 +59,55 @@ export default class MetadataQueryAPIHelper {
         field: string,
         oldValue: ?MetadataFieldValue,
         newValue: ?MetadataFieldValue,
-    ): JSONPatchOperations => [
-        {
+    ): JSONPatchOperations => {
+        let operation = JSON_PATCH_OP_REPLACE;
+
+        if (isNil(oldValue) && newValue) {
+            operation = JSON_PATCH_OP_ADD;
+        }
+
+        if (oldValue && isNil(newValue)) {
+            operation = JSON_PATCH_OP_REMOVE;
+        }
+
+        const testOp = {
             op: JSON_PATCH_OP_TEST,
             path: `/${field}`,
             value: oldValue,
-        },
-        {
-            op: JSON_PATCH_OP_REPLACE,
+        };
+        const patchOp = {
+            op: operation,
             path: `/${field}`,
             value: newValue,
-        },
-    ];
+        };
+
+        if (operation === JSON_PATCH_OP_REMOVE) {
+            delete patchOp.value;
+        }
+
+        const ops = operation === JSON_PATCH_OP_ADD ? [patchOp] : [testOp, patchOp];
+        return ops;
+    };
+
+    getMetadataQueryFields = (): string[] => {
+        /*
+            Example metadata query:
+            const query = {
+                from: 'enterprise_12345.myAwesomeTemplateKey',
+                fields: [
+                    'name', // base representation field for an item (name, size, etag etc.)
+                    'metadata.enterprise_12345.myAwesomeTemplateKey.field_1', // metadata instance field
+                    'metadata.enterprise_12345.myAwesomeTemplateKey.field_2', // metadata instance field
+                    'metadata.enterprise_12345.myAwesomeTemplateKey.field_3' // metadata instance field
+                ],
+                ancestor_folder_id: 0,
+            };
+
+            This function will return ['field_1', 'field_2', 'field_3']
+        */
+        const { fields = [], from } = this.metadataQuery;
+        return fields.filter(field => field.includes(from)).map(field => field.split(`${from}.`).pop());
```

Comment:
Minor: would `field.split('.').pop()` work, since the field name always appears after the last dot?